app/src/test/java/com/duckduckgo/app/sitepermissions/SitePermissionsViewModelTest.kt

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.sitepermissions
+
+import app.cash.turbine.test
+import com.duckduckgo.app.CoroutineTestRule
+import com.duckduckgo.app.browser.R
+import com.duckduckgo.app.location.GeoLocationPermissions
+import com.duckduckgo.app.location.data.LocationPermissionEntity
+import com.duckduckgo.app.location.data.LocationPermissionType
+import com.duckduckgo.app.location.data.LocationPermissionsRepositoryAPI
+import com.duckduckgo.app.settings.db.SettingsDataStore
+import com.duckduckgo.app.sitepermissions.SitePermissionsViewModel.Command.LaunchWebsiteAllowed
+import com.duckduckgo.app.sitepermissions.SitePermissionsViewModel.Command.ShowRemovedAllConfirmationSnackbar
+import com.duckduckgo.app.statistics.pixels.Pixel
+import com.duckduckgo.site.permissions.impl.SitePermissionsRepository
+import com.duckduckgo.site.permissions.store.sitepermissions.SitePermissionsEntity
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.runTest
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+
+@ExperimentalCoroutinesApi
+class SitePermissionsViewModelTest {
+
+    @ExperimentalCoroutinesApi
+    @get:Rule
+    var coroutineRule = CoroutineTestRule()
+
+    private lateinit var viewModel: SitePermissionsViewModel
+
+    private val mockSitePermissionsRepository: SitePermissionsRepository = mock()
+    private val mockLocationPermissionsRepository: LocationPermissionsRepositoryAPI = mock()
+    private val mockGeoLocationPermissions: GeoLocationPermissions = mock()
+    private val mockSettingsDataStore: SettingsDataStore = mock()
+    private val mockPixel: Pixel = mock()
+
+    @Before
+    fun before() {
+        viewModel = SitePermissionsViewModel(
```

Comment:
ditto.

Reply:
In this case I will keep it because it will introduce much more lines of code and the logic is shared between all tests. I will change the constructor to the variable declaration though 👍